lib/appsignal/rack/hanami_instrumentation.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require "rack"
+
+module Appsignal
+  # @api private
+  module Rack
+    class HanamiInstrumentation
+      def initialize(app, options = {})
+        Appsignal.logger.debug "Initializing Appsignal::Rack::HanamiInstrumentation"
+        @app = app
+        @options = options
+      end
+
+      def call(env)
+        if Appsignal.active?
+          call_with_appsignal_monitoring(env)
+        else
+          @app.call(env)
+        end
+      end
+
+      def call_with_appsignal_monitoring(env)
+        params = ::Hanami::Action::BaseParams.new(env)
+        request = ::Hanami::Action::Request.new(
+          :env => env,
+          :params => params,
+          :sessions_enabled => true
+        )
+
+        transaction = Appsignal::Transaction.create(
+          SecureRandom.uuid,
+          Appsignal::Transaction::HTTP_REQUEST,
+          request
+        )
+
+        begin
+          Appsignal.instrument("process_action.hanami") do
+            @app.call(env)
+          end
+        rescue Exception => error # rubocop:disable Lint/RescueException
+          transaction.set_error(error)
+          raise error
+        ensure
+          transaction.params = request.params.to_h
+          transaction.set_action_if_nil("#{request.request_method} #{request.path}")
```

Comment:
The action name should have low cardinality -- meaning it can be used to group items. [We already discussed this on Slack, just leaving the comment here so we don't forget]

So the action name would need to be something like `GET /books/:id` or `Actions::Books::Show` -- but not `GET /books/123`.

Reply:
Waiting for some feedback here: https://gitter.im/hanami/chat?at=63a06fbc69ce3c5338359379

If we can't find a solution, we should start instrumenting by framework component instead of Rack.